docs/Growing.md

```diff
@@ -117,6 +117,139 @@ functionality and then moves on to producing code that passes these tests. There
 having good test coverage and code being designed for testability, which will in turn often prevent future changes from
 breaking old code. Much of mojo.js was developed using TDD.
 
+## Prototype
+
+Mojolicious allows you to both create a lite app and a full app. The lite app is perfect for rapid prototyping.
+
+You can create the lite app by:
+
+```
+  $ npx mojo create-lite-app
+```
+
+This creates a single file.
+
+```
+index.js  // Templates can be inlined
+```
+
+`index.js` starts the mojo app server by default on port 3000 and renders a minimal webpage.
+
+You can test simply by opening a web browser at url `localhost:3000`.
+
+Full Mojolicious applications on the other hand are set-up to be a well organized distribution to maximize 
+maintainability.
+
+First, create your project folder and step into it. 
+
+```
+$ mkdir my_app
+$ cd my_app
+```
+
+You can now create the full app by:
+
+```
+$ npx mojo create-full-app
+```
+
+The full app will have the following structure:
+
+```
+my_app                                // Application dir created manually
+|- config.yml                         // Configuration file
+|- index.js                           // Application script
+|- node_modules/
+|   |- *lots of node files*
+|- package.json                       // Node package information and settings
+|- package-lock.json                  // Describes exact tree generated in node_modules/
+|- test/                              // Test directory
+|   |- example.js                     // Random test
+|- controllers/                       // Controller directory
+|   |- example.js                     // 
+|- models/                            // Model directory
+|- public/                            // Static file directory (served automatically)
+|   |- index.html                     // Static HTML file
+|- views/                             // Views directory
+|   |- example/                       // View directory for "Example"
+|   |   |- welcome.html.ejs
+|   |- layouts/                       // View directory for layouts
+|   |   |- default.html.ejs           // Layout view/template
+
+```
+
+Again, both application skeletons can be automatically generated with:
+
+```
+$ npx mojo create-lite-app
+$ npx mojo create-full-app
+```
+
+Feature-wise both are almost equal, the only real differences are organizational, so each one can be gradually 
+transformed into the other.
+
+### Foundation
+
+We start our new application with a single javascript script.
+
+```
+$ mkdir my_app
+$ cd my_app
+$ touch myapp.pl
+```
+
+This will be the foundation for our login manager example application.
+
+```
+import mojo from '@mojojs/core';
+
+const app = mojo();
+
+app.get('/', ctx => ctx.render({text: 'I ♥ Mojo!'}));
+
+app.start();
+```
+
+The built-in development web server makes working on your application a lot of fun thanks to automatic reloading.
```

Comment:
As far as I know mojo.js does not include a auto-reloader like morbo. We should recommend `nodemon` for that.
```suggestion
Now that we have a simple webapp, let's start it as a server with node
```